### PR TITLE
[Java] Implement concept exercise conditionals-ternary

### LINKED ISSUE #1869: [Java] Implement new Concept Exercise: conditionals-ternary
This issue describes how to implement the `conditionals-ternary` concept exercise for the Java track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how and when to use ternary expressions  and the `?:` operator in Java.

## Learning objectives

- Know how to use ternary expressions - using the `?:` operator.

## Out of scope

Nothing notable

## Concepts

- ternary expressions

## Prerequisites

- `booleans`
- `conditionals-if`

### Hints

Nothing to report

### After

Nothing to report

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/java/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[analyzer]: https://github.com/exercism/java-analyzer
[representer]: https://github.com/exercism/java-representer

### PR DESCRIPTION
**Pull request by TalesDias**
closes #1869

### CODE REVIEW

**Comment by TalesDias** [comment]
Took the liberty to create a new story for this one, hoping you could comment on it as well

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+In this exercise, you'll be implementing rules for calculating the total salary of a empyoyee in a month. The International Siderurgy Company (ISC) needs help to calculate the salary for the employees, given that different factors can alter the final wage value for each employee.
+
+You have the following tasks
+
+## 1. Decide a penalty for days skipped needs to be applied
+
+Implement the `multiplierPerDaysSkipped` method that returns the penalty multiplier based on the days the employee skipped the job. A 15% penalty is applied if more than five days wore skipped.
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:7]:
```suggestion
Implement the `multiplierPerDaysSkipped` method that returns the salary multiplier based on the number of days the employee skipped the job. A 15% penalty is applied if more than five days were skipped.
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## 2. Calculate the bonus for products sold
+
+Implement the `multiplierPerProductsSold` and `bonusProductSold` methods. The ISC pays ten monetary units for each product sold, but if the employee sold more than twenty products, the multiplier is improved to thirteen. `multiplierPerProductsSold` should decide which multiplier is applied and `bonusProductSold` should return the total bonus in monetary units.
+
+```java
+multiplierPerProductsSold(25);
+// => 13
+
+bonusProductSold(5);
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:25]: Maybe rename `bonusProductSold` to `bonusForProductsSold`? Just to be a little bit more descriptive.

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+In this exercise, you'll be implementing rules for calculating the total salary of a empyoyee in a month. The International Siderurgy Company (ISC) needs help to calculate the salary for the employees, given that different factors can alter the final wage value for each employee.
+
+You have the following tasks
+
+## 1. Decide a penalty for days skipped needs to be applied
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:5]: Maybe rephrase this slightly as students are returning a multiplier, and not a penalty.

```suggestion
## 1. Determine the salary multiplier
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+multiplierPerProductsSold(25);
+// => 13
+
+bonusProductSold(5);
+// => 50
+```
+
+## 3. Calculate the final salary for the employee
+
+Implement the `finalSalary` method. It should be able to multiply the base salary by the penalty and sum the bonus and return the result, but keep in mind that salaries should be capped at the maximum value
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:31]: > multiply the base salary by the penalty

Maybe make this `multiply the base salary by the salary multiplier`? See my previous comment above,

Comment (ErikSchierboom) [reply languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:31]: > keep in mind that salaries should be capped at the maximum value

I don't see the actual maximum value listed anywhere. Could you add that?

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+finalSalary(0,0);
+// => 1000
+
+finalSalary(0,100);
+// => 2000
+```
+
+### Attention
+
+You should not use _if-else_ statements in this exercise, only the _ternary operator_ is allowed
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:43]: It will be entirely possible that students will start coding once they've read one instruction. I'd remove this section and either add something like this to each task, or perhaps add it to the "You have the following tasks" line.

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+bonusProductSold(5);
+// => 50
+```
+
+## 3. Calculate the final salary for the employee
+
+Implement the `finalSalary` method. It should be able to multiply the base salary by the penalty and sum the bonus and return the result, but keep in mind that salaries should be capped at the maximum value
+
+```java
+finalSalary(0,0);
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:34]:
```suggestion
finalSalary(0, 0);
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+// => 1000
+
+finalSalary(0,100);
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:37]:
```suggestion
finalSalary(0, 100);
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md (hunk 6 above)
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:34]: Which parameter represents which value? I would suggest being explicit about the parameter values in both the instructions above and in this bit of code (see [this example](https://github.com/exercism/v3/blob/master/languages/java/exercises/concept/conditionals/.docs/instructions.md#3-implement-the-decision-logic-for-hand-scores-larger-than-20-points)).

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    } else{
+        if (expr3) {
+            return val2;
+        } else {
+            return val3;
+        }
+    }
+} else {
+    return val4;
+}
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/after.md:25]: I'm no Java expert, but I find this not easier to read :)

```suggestion
// easier to read
if (expr1){
    if (expr2){
        return val1;
    } else{
        if (expr3) {
            return val2;
        } else {
            return val3;
        }
    }
} else {
    return val4;
}
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
+_ternary operators_ can be thought of as being _if-else_ inline. It is a good example that most of the times you have different ways of achieving the same result.
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/after.md:1]: The `after.md` document should contain all information that is introduced in the .docs/introduction.md document. See [the docs](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#docsaftermd). It doesn't necessarily have to use the exact same wording, but it feels like this document is missing the introduction bit.

**Review by ErikSchierboom — COMMENTED** [review]
Thanks for working on this! I've left some comments.

Also, could you add an entry to the [`config.json` file](https://github.com/exercism/v3/blob/master/languages/java/config.json)?

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md (hunk 9 above)
Comment (TalesDias) [reply languages/java/exercises/concept/conditionals-ternary/.docs/after.md:1]: Oh, i didn't remember that, i will modify it

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md (hunk 8 above)
Comment (TalesDias) [reply languages/java/exercises/concept/conditionals-ternary/.docs/after.md:25]: Unfortunately i understand you, i think most of this is due to the brackets.
I think a shallow nested if will do the trick

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md (hunk 3 above)
Comment (TalesDias) [reply languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:5]: Makes sense, while writing i was really thinking that `penalty` was a strong term, but couldn't find a nicer one.

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md (hunk 8 above)
Comment (ErikSchierboom) [reply languages/java/exercises/concept/conditionals-ternary/.docs/after.md:25]: 👍

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md (hunk 9 above)
Comment (ErikSchierboom) [reply languages/java/exercises/concept/conditionals-ternary/.docs/after.md:1]: Great!

**Comment by ErikSchierboom** [comment]
You can re-request a review from me when you're done by clicking on the spinner icon to the right of my name in the reviewers list.

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md (hunk 2 above)
Comment (TalesDias) [reply languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:25]: Agreed

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md (hunk 5 above)
Comment (TalesDias) [reply languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:43]: I like the idea of adding it to the beginning, maybe something like:
> You have three tasks, but keep in mind that _if/else_ statements should not be used.
or 
>  You have three tasks, use just _ternary operators_  to solve them

File: languages/java/exercises/concept/conditionals-ternary/.meta/design.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
+## Learning objectives
```
[the remark this answers is not in the document]
Comment (TalesDias) [reply languages/java/exercises/concept/conditionals-ternary/.meta/design.md:1]: Never used analysers before, but the sections is now there  : )

File: languages/java/exercises/concept/conditionals-ternary/src/main/java/SalaryCalculator.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
+public class SalaryCalculator {
+
+    private final double BASESALARY = 1000.0;
+    private final double MAXIMUMSALARY = 2000.0;
```
[the remark this answers is not in the document]
Comment (TalesDias) [reply languages/java/exercises/concept/conditionals-ternary/src/main/java/SalaryCalculator.java:4]: I thought they would not know what the `base salary` and `maximum salary`, but now the instructions cover it well.

[the remark this answers is not in the document]
Comment (TalesDias) [reply languages/java/exercises/concept/conditionals-ternary/src/main/java/SalaryCalculator.java:4]: And probably the new keywords could confuse them, is better to just remove  it

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md (hunk 8 above)
Comment (TalesDias) [reply languages/java/exercises/concept/conditionals-ternary/.docs/after.md:25]:
```java
// hard to read
int value = expr1 ? expr2 && expr3 ? val1 : (val2 + val3) : val4;

// easier to read
if (expr1){

    if (expr2 && expr3){

        return val1;
    } else {

        return val2 + val3;
    }

} else {
    
    return val4;
}

 ```

File: languages/java/config.json [hunk 12]
```diff
@@ -57,6 +57,12 @@
         "uuid": "2165c331-6b0c-49b8-9555-bad577ec4fc6",
         "concepts": ["strings"],
         "prerequisites": ["basics"]
+      },
+      {
+        "slug": "conditionals-ternary",
```
Comment (ErikSchierboom) [inline languages/java/config.json:62]: The name of the exercise implies a hierarchy: conditions -> ternary. As per [the docs](https://github.com/exercism/v3/blob/master/docs/maintainers/determining-concepts.md#naming-concepts), exercise names should not use a hierarchy. I think a better name in this game would be `ternary-operators`. Note that this would also mean changing the directory name of the exercise.

File: languages/java/config.json [hunk 13]
```diff
@@ -57,6 +57,12 @@
...
+        "uuid": "2165c331-6b0c-49b8-9555-bad577ec4fc6",
+        "concepts": ["strings"],
```
Comment (ErikSchierboom) [inline languages/java/config.json:64]: This key should be the concepts this exercise unlocks:

```suggestion
        "concepts": ["ternary-operators"],
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md [hunk 14]
```diff
@@ -0,0 +1,52 @@
+## Ternary Operator
+
+The _ternary operators_ can be thought of as being a compact version of _if-else_. It's usually used in (but not restricted to) return statements, it needs just one single to make the decision, returning the left value if the expression is `true` and the right value if `false`.
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/after.md:3]: I think a word might be missing here.

```suggestion
The _ternary operators_ can be thought of as being a compact version of _if-else_. It's usually used in (but not restricted to) return statements, it needs just one single line to make the decision, returning the left value if the expression is `true` and the right value if `false`.
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+
+// is equivalent to this if/else
+if ( 5 > 4 ) {
+    return true;
+} else {
+    return false;
+}
+```
+
+So, how to decide between _if-else_ and _ternary_ ? Well, _Ternary_ is used in simple cenarios, where you just need to return a value based on a condition and no extra computation is needed, use _if-else_ for everthing else, like nested conditions, big expressions and when more than one line is needed to decide the return value.
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/after.md:20]:
```suggestion
So, how to decide between _if-else_ and _ternary_ ? Well, _ternary operators_ are used in simple scenarios, where you just need to return a value based on a condition and no extra computation is needed. Use an _if-else_ for everthing else, like nested conditions, big expressions and when more than one line is needed to decide the return value.
```

Comment (ErikSchierboom) [reply languages/java/exercises/concept/conditionals-ternary/.docs/after.md:20]: This section is great! It is sections like this that are perfect for an after.md document.

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+
+And don't abuse of nested _ternary operators_ as the expression becomes confusing, prefer nested if's to handle those cases.
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/after.md:22]: Maybe rephrase this slightly? Abuse sounds a bit strong IMHO.

```suggestion
While you can nest _ternary operators_, the code often becomes hard to read. In these cases, nested if's are preferred.
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+
+```java
+
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/after.md:25]:
```suggestion
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+    } else {
+
+        return val2 + val3;
+    }
+
+} else {
+
+    return val4;
+}
+
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/after.md:44]:
```suggestion
if (expr1){
    if (expr2 && expr3){
        return val1;
    } else {
        return val2 + val3;
    }
} else {
    return val4;
}
```

Comment (ErikSchierboom) [reply languages/java/exercises/concept/conditionals-ternary/.docs/after.md:44]: I don't know if this is idiomatic in Java, but you could consider promoting an early return variant:

```java
if (!expr1){
    return val4;
}

if (expr2 && expr3){
    return val1;
}

return val2 + val3;
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+_Ternary_ and _if/else_ statements are a good example that you have different ways of achieving the same result when programming.
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/after.md:47]:
```suggestion
_Ternary operators_ and _if/else_ statements are a good example that you have different ways of achieving the same result when programming.
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md [hunk 20]
```diff
@@ -0,0 +1,52 @@
+## Ternary Operator
+
+The _ternary operators_ can be thought of as being a compact version of _if-else_. It's usually used in (but not restricted to) return statements, it needs just one single to make the decision, returning the left value if the expression is `true` and the right value if `false`.
+
+A lot of simple _if/else_ expressions can be simplified using _ternary operators_
+
+```java
+
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/after.md:8]:
```suggestion
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md [hunk 21]
```diff
@@ -0,0 +1,52 @@
+## Ternary Operator
+
+The _ternary operators_ can be thought of as being a compact version of _if-else_. It's usually used in (but not restricted to) return statements, it needs just one single to make the decision, returning the left value if the expression is `true` and the right value if `false`.
+
+A lot of simple _if/else_ expressions can be simplified using _ternary operators_
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/after.md:5]:
```suggestion
A lot of simple _if/else_ expressions can be simplified using _ternary operators_.
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+```java
+
+int daysSkipped = 3;
+multiplierPerDaysSkipped(daysSkipped);
+// => 1
+
+daysSkipped = 7;
+multiplierPerDaysSkipped(daysSkipped);
+// => 0.85
+
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:18]:
```suggestion
int daysSkipped = 3;
multiplierPerDaysSkipped(daysSkipped);
// => 1

daysSkipped = 7;
multiplierPerDaysSkipped(daysSkipped);
// => 0.85
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+## 2. Calculate the bonus for products sold
+
+Implement the `multiplierPerProductsSold` and `bonusForProductSold` methods. The ISC pays ten monetary units for each product sold, but if the employee sold more than twenty products, the multiplier is improved to thirteen. `multiplierPerProductsSold` should decide which multiplier is applied and `bonusForProductSold` should return the total bonus in monetary units.
+
+```java
+
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:26]:
```suggestion
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+Implement the `finalSalary` method. It should be able to multiply the base salary of 1000.00 by the salary multiplier and sum the bonus and return the result, but keep in mind that salaries should be capped at 2000.00;
+
+```java
+
+int daysSkipped = 2;
+int productsSold = 3;
+finalSalary(daysSkipped, productsSold);
+// => 1030
+
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:46]:
```suggestion
int daysSkipped = 2;
int productsSold = 3;
finalSalary(daysSkipped, productsSold);
// => 1030
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/introduction.md [hunk 25]
```diff
@@ -0,0 +1,28 @@
+## Ternary Operator
+
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/introduction.md:2]: The introduction document doesn't have a heading.

```suggestion
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/introduction.md [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+The _ternary operator_ is a lightweight, compact alternative for simple _if/else_ statements. Usually used in (but not restricted to) return statements, it needs just one single to make the decision, returning the left value if the expression is `true` and the right value if `false`, as follows:
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/introduction.md:3]:
```suggestion
The _ternary operator_ is a lightweight, compact alternative for simple _if/else_ statements. Usually used in (but not restricted to) return statements, it needs just one single line to make the decision, returning the left value if the expression is `true` and the right value if `false`, as follows:
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/introduction.md [hunk 27]
```diff
@@ -0,0 +1,28 @@
...
+
+```java
+
+boolean expr = 0 != 200;
+
+int value = expr ? 22 : 33;
+// => 22
+
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/introduction.md:11]:
```suggestion
boolean expr = 0 != 200;

// Ternary statement
int value = expr ? 22 : 33;
// => 22
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/introduction.md [hunk 28]
```diff
@@ -0,0 +1,28 @@
...
+return 5 > 4 ? true : false;
+
+// is equivalent to this if/else
+if ( 5 > 4 ) {
+    return true;
+} else {
+    return false;
+}
+
+```
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/introduction.md:28]: One of our goals with the `introduction.md` document is to have them be very concise (see [the docs](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#docsintroductionmd)). As the above example introduces the ternary statement syntax, this example can be removed:

```suggestion
```

File: languages/java/config.json [hunk 29]
```diff
@@ -57,6 +57,12 @@
...
+        "prerequisites": ["conditionals", "numbers"]
```
Comment (ErikSchierboom) [inline languages/java/config.json:65]: The `design.md` document lists `conditionals-if` as the prerequiste (which has the same naming problem I mentioned above by the way), so I think this should be:

```suggestion
        "prerequisites": ["conditionals-if", "numbers"]
```

File: languages/java/exercises/concept/conditionals-ternary/.meta/design.md [hunk 30]
```diff
@@ -0,0 +1,24 @@
...
+This exercise's prerequisites Concepts are:
+
+- `numbers`: double and int types
+- `conditionals-if` : use `if else` expressions
+
+## Analyzer
+
+This exercise could have the following rule added to the analyzer:
+
+- Verify that none _if/else_ statments were used.
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.meta/design.md:24]:
```suggestion
- Verify that ternary operators were used.
```

File: languages/java/exercises/concept/conditionals-ternary/src/main/java/SalaryCalculator.java [hunk 31]
```diff
@@ -0,0 +1,20 @@
...
+        throw new UnsupportedOperationException("Please implement the SalaryCalculator.multiplierPerProductsSold() method");
+    }
+
+    public double bonusForProductSold (int productsSold) {
+        throw new UnsupportedOperationException("Please implement the SalaryCalculator.bonusForProductSold() method");
+    }
+
+    public double finalSalary (int daysSkipped, int productsSold) {
+        throw new UnsupportedOperationException("Please implement the SalaryCalculator.finalSalary() method");
+
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/src/main/java/SalaryCalculator.java:17]:
```suggestion 
```

File: languages/java/exercises/concept/conditionals-ternary/src/main/java/SalaryCalculator.java [hunk 32]
```diff
@@ -0,0 +1,20 @@
...
+    public double bonusForProductSold (int productsSold) {
+        throw new UnsupportedOperationException("Please implement the SalaryCalculator.bonusForProductSold() method");
+    }
+
+    public double finalSalary (int daysSkipped, int productsSold) {
+        throw new UnsupportedOperationException("Please implement the SalaryCalculator.finalSalary() method");
+ 
+    } 
+
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/src/main/java/SalaryCalculator.java:19]:
```suggestion
```

File: languages/java/exercises/concept/conditionals-ternary/src/main/java/SalaryCalculator.java [hunk 33]
```diff
@@ -0,0 +1,20 @@
+public class SalaryCalculator {
+
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/src/main/java/SalaryCalculator.java:2]:
```suggestion
```

File: languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md [hunk 34]
```diff
@@ -0,0 +1,47 @@
+In this exercise, you'll be implementing rules for calculating the total salary of a empyoyee in a month. The International Siderurgy Company (ISC) needs help to calculate the salary for the employees, given that different factors can alter the final wage value for each employee.
+
+You have three tasks, but keep in mind that if/else statements should not be used.
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/instructions.md:3]: Maybe make it slightly more explicit what is expected:

```suggestion
You have three tasks and you should use the ternary operator instead of if/else statements to implement them.
```

**Review by ErikSchierboom — APPROVED** [review]
Nice! I've left some additional comments. I've approved this PR. You can either merge this PR and fix the comments in a follow-up PR, or fix them in this PR first. The choice is up to you :)

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md (hunk 16 above)
Comment (TalesDias) [reply languages/java/exercises/concept/conditionals-ternary/.docs/after.md:22]: of course, hahha

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md (hunk 18 above)
Comment (TalesDias) [reply languages/java/exercises/concept/conditionals-ternary/.docs/after.md:42]: I don't usually see this, but i think the first expression will already be enough
Do you think it's clear to the student why we can remove the _else's_?

File: languages/java/config.json (hunk 12 above)
Comment (ErikSchierboom) [inline languages/java/config.json:62]: The slug should also be updated:

```suggestion
        "slug": "ternary-operators",
```

Comment (ErikSchierboom) [reply languages/java/config.json:62]: Note that this also means renaming the exercise directory to `ternary-operators`.

File: languages/java/exercises/concept/conditionals-ternary/.docs/after.md (hunk 18 above)
Comment (ErikSchierboom) [reply languages/java/exercises/concept/conditionals-ternary/.docs/after.md:42]: Okay, let's go with just the first version.

File: languages/java/exercises/concept/conditionals-ternary/.docs/introduction.md [hunk 35]
```diff
@@ -0,0 +1,11 @@
...
+
+```java
+boolean expr = 0 != 200;
+
+// Ternary statement
+int value = expr ? 22 : 33;
+// => 22
+```
+
+A lot of simple _if/else_ expressions can be simplified using _ternary operators_
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.docs/introduction.md:11]:
```suggestion
A lot of simple _if/else_ expressions can be simplified using _ternary operators_.
```

File: languages/java/exercises/concept/conditionals-ternary/.meta/design.md [hunk 36]
```diff
@@ -0,0 +1,24 @@
...
+
+- Know how to use ternary exepressions - using the `?:` operator.
+
+## Out of scope
+
+- Nothing notable
+
+## Concepts
+
+- ternary expressions
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/conditionals-ternary/.meta/design.md:11]: This should list the actual concept string that is being unlocked:

```suggestion
- `ternary-operators`: ternary expressions
```

**Review by ErikSchierboom — APPROVED** [review]
Just a couple of comments :)

File: languages/java/exercises/concept/ternary-operators/.meta/design.md [hunk 37]
```diff
@@ -0,0 +1,24 @@
...
+
+- Know how to use ternary exepressions - using the `?:` operator.
+
+## Out of scope
+
+- Nothing notable
+
+## Concepts
+
+- ternary operators
```
Comment (ErikSchierboom) [inline languages/java/exercises/concept/ternary-operators/.meta/design.md:11]:
```suggestion
- `ternary-operators`: how to use the ternary operator to do conditional logic
```

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
@TalesDias ship it! 🎉 (our convention is for PR submitters to merge themselves)

**Comment by ErikSchierboom** [comment]
@TalesDias Please use the Squash and Merge option

**Comment by TalesDias** [comment]
> @TalesDias Please use the Squash and Merge option

How exactly i do this? i just have a message: "Only those with write access to this repository can merge pull requests."

**Comment by ErikSchierboom** [comment]
Ah sorry about that. I've merged! 🎉  Thanks for adding this!
